src/Monad/Listt/functions.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Widmogrod\Monad\Listt;
+
+use Widmogrod\Functional as f;
+use Widmogrod\Monad\Maybe as m;
+use Widmogrod\Primitive\Listt;
+
+/**
+ * @var callable
+ */
+const mempty = 'Widmogrod\Monad\Listt\mempty';
+
+/**
+ * Get an empty Listt.
+ *
+ * @return Listt
+ */
+function mempty(): Listt
+{
+    return Listt::mempty();
+}
+
+/**
+ * @var callable
+ */
+const push = 'Widmogrod\Monad\Listt\push';
+
+/**
+ * push :: [a] -> [a] -> [a]
+ *
+ * Append Listt with values.
+ *
+ * @param Listt    $list
+ * @param iterable $values
+ *
+ * @return Listt
+ *
+ * @throws \Widmogrod\Primitive\TypeMismatchError
+ */
+function push(Listt $list, iterable $values): Listt
+{
+    return $list->concat(fromRecursive($values));
```

Comment:
Another thins isn't this to much responsibility of this function?
- concat two lists
- concat and flat map two list recursively 

Maybe it make sense to split those responsibilities because sometimes for given set of data 
```php
$a = Listt::of([1,2, 3]);
$b = Listt::of([[4],[5,[6,7]]])
```

I may want to have this behaviour
```php
assert(push($a, $b) === Listt::of([1,2,3,4,5,6]))
``` 

Or this
```php
assert(push($a, $b) === Listt::of([1,2,3,[4],[5,[6,7]]))
``` 

But most time I wan't to have a control of what to use.

WDYT?